src/object.c

```diff
@@ -989,6 +989,9 @@ int equalStringObjects(robj *a, robj *b) {
          * long is the same. */
         return a->ptr == b->ptr;
     } else {
+        if (stringObjectLen(a) != stringObjectLen(b)) {
```

Comment:
if either a or b is INT encoding, we will have to call sdigits10().
In this case, I think we shouldn't compare them here.

Reply:
You’re right, fixed it.